### PR TITLE
allow force debug configuration + add service full path config in json

### PR DESCRIPTION
**Pull request by berryve**
* When debug attribute in the AppsTransferButton or AppsDiscoveryButton is set to true, the android_debug.json file will be used instead of the production one
* Add configuration for the full path of the SendKinService in the configuration of the app in the json file.
* Fix SendKinService params
* Fixes in tests & build errors

### CODE REVIEW

File: transfer/src/main/java/org/kinecosystem/transfer/repositories/EcosystemAppsRepository.kt [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
-
     companion object {
         private var instance: EcosystemAppsRepository? = null
 
-        fun getInstance(currentPackage: String, local: EcosystemAppsLocalRepo, remote: EcosystemAppsRemoteRepo, uiHandler: Handler): EcosystemAppsRepository {
+        fun getInstance(context: Context): EcosystemAppsRepository {
             if (instance == null) {
-                instance = EcosystemAppsRepository(currentPackage, local, remote, uiHandler)
+                instance = EcosystemAppsRepository(context.packageName, EcosystemAppsLocalRepo(context),
+                        EcosystemAppsRemoteRepo(), Handler(Looper.getMainLooper()))
```
Comment (shayb) [inline transfer/src/main/java/org/kinecosystem/transfer/repositories/EcosystemAppsRepository.kt:18]: i thought the handler will be outside param so we can replace it on testing with some testing component and not hard coded mainlooper

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/customview/AppsTransferButton.kt [hunk 2]
```diff
@@ -11,16 +11,36 @@ import org.kinecosystem.appstransfer.view.AppsTransferActivity
 class AppsTransferButton @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
         TextView(context, attrs, defStyleAttr) {
 
-    private fun startAppsDiscoveryActivity() {
-        context.startActivity(AppsTransferActivity.getIntent(context))
+    private fun startAppsDiscoveryActivity(debugMode: Boolean) {
```
Comment (shayb) [inline appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/customview/AppsTransferButton.kt]: pls use default as false

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt [hunk 3]
```diff
@@ -70,13 +72,15 @@ class AppsTransferActivity : AppCompatActivity(), IAppsTransferView {
...
         super.onCreate(savedInstanceState)
+        processIntent()
         setContentView(R.layout.apps_transfer_activity)
         dataGroup = findViewById(R.id.data)
         noDataGroup = findViewById(R.id.noData)
         loader = findViewById(R.id.loader)
         list = findViewById(R.id.list)
         list.clickListener = presenter
-        presenter = AppsTransferPresenter(TransferManager(this), EcosystemAppsLocalRepo(this))
+        presenter = AppsTransferPresenter(TransferManager(this),
```
Comment (shayb) [inline appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt]: is something changed here?

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/SenderServiceBinder.kt [hunk 4]
```diff
@@ -77,7 +78,7 @@ class SenderServiceBinder(private val context: Context?) {
                     kinTransferComplete?.let {
                         updateTransferCompleted(it)
                     } ?: kotlin.run {
-                        startSendKinAsnyc(receiverAddress, amount, memo, object : KinTransferCallback {
+                        startSendKinAsync(senderAppName, senderAppId, receiverAddress, amount, memo, object : KinTransferCallback {
```
Comment (berryve) [inline appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/SenderServiceBinder.kt]: that's not right - fixing it ...

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/SenderServiceBinder.kt [hunk 5]
```diff
@@ -97,34 +98,38 @@ class SenderServiceBinder(private val context: Context?) {
...
-            intent.component = ComponentName(senderPackageName, serviceFullPath)
-            intent.`package` = senderPackageName
-            context?.packageManager?.let {
-                val resolveInfos: MutableList<ResolveInfo> = it.queryIntentServices(intent, 0)
-                if (!resolveInfos.any()) {
-                    throw ServiceConfigurationException(serviceFullPath, "Service not found - Service must be implemented")
-                }
-                if (resolveInfos.filter { it.serviceInfo.exported }.any()) {
-                    throw ServiceConfigurationException(serviceFullPath, "Service should not be exported")
+            context?.let {
```
Comment (shayb) [inline appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/SenderServiceBinder.kt:112]: why did you add this check if  - its null there is no - else then... some error

File: transfer/src/main/java/org/kinecosystem/transfer/repositories/EcosystemAppsRepository.kt (hunk 1 above)
Comment (berryve) [reply transfer/src/main/java/org/kinecosystem/transfer/repositories/EcosystemAppsRepository.kt:18]: could create an additional getMockInstance method or in any case find a different way of managing this.
It doesn't make sense that each time we call get instance we create new instances of the repos that are not used!

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/view/AppInfoActivity.kt [hunk 6]
```diff
@@ -189,16 +191,16 @@ class AppInfoActivity : AppCompatActivity(), IAppInfoView {
             finish()
         }
         with(app?.metaData?.experienceData) {
-            findViewById<TextView>(R.id.howInfo).text = this?.howTo ?: ""
-            findViewById<TextView>(R.id.experienceName).text = this?.name ?: ""
-            findViewById<TextView>(R.id.experienceInfo).text = this?.about ?: ""
+            (findViewById<TextView>(R.id.howInfo) as TextView).text = this?.howTo ?: ""
```
Comment (shayb) [inline appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/view/AppInfoActivity.kt]: you dont need to cast if - <TextView> is there

**Review by shayb — COMMENTED** [review]
pls look at my comments

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt (hunk 3 above)
Comment (berryve) [reply appsTransfer/src/main/java/org/kinecosystem/appstransfer/view/AppsTransferActivity.kt]: no just unintentional new line

File: transfer/src/main/java/org/kinecosystem/transfer/sender/service/SendKinServiceBase.java [hunk 7]
```diff
@@ -16,9 +16,9 @@
 
     public void cancelCallback() { transferKinCallback = null; }
 
-    public abstract KinTransferComplete transferKin(@NonNull final String senderAppName, @NonNull final String senderAppId, @NonNull final String toAddress, final int amount, @NonNull final String memo) throws KinTransferException;
+    public abstract KinTransferComplete transferKin(@NonNull final String senderAppId, @NonNull final String senderAppName, @NonNull final String toAddress, final int amount, @NonNull final String memo) throws KinTransferException;
```
Comment (berryve) [inline transfer/src/main/java/org/kinecosystem/transfer/sender/service/SendKinServiceBase.java]: that's not right either

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/view/customView/AppsDiscoveryButton.kt [hunk 8]
```diff
@@ -11,16 +11,36 @@ import org.kinecosystem.appsdiscovery.view.AppsDiscoveryActivity
 class AppsDiscoveryButton @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) :
         TextView(context, attrs, defStyleAttr) {
 
-    private fun startAppsDiscoveryActivity() {
-        context.startActivity(AppsDiscoveryActivity.getIntent(context))
+    private fun startAppsDiscoveryActivity(debugMode: Boolean) {
```
Comment (shayb) [inline appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/view/customView/AppsDiscoveryButton.kt]: probably this is wrong to be private 
need to change to public and debug should have default false

File: appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/SenderServiceBinder.kt (hunk 5 above)
Comment (berryve) [reply appsTransfer/src/main/java/org/kinecosystem/appstransfer/presenter/SenderServiceBinder.kt:112]: there was no else before either ;-) 
Anyway this is called at onResume - chances of the context being null are zero.
I needed to pass a non-optional to the getInstance method

File: appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/view/customView/AppsDiscoveryButton.kt (hunk 8 above)
Comment (berryve) [reply appsDiscovery/src/main/java/org/kinecosystem/appsdiscovery/view/customView/AppsDiscoveryButton.kt]: method is actually not necessary
